helm-control

```diff
@@ -66,64 +73,70 @@ static const int32_t reverseSensorSteps = (reverseMinimum - reverseMaximum);
 static const int32_t forwardSensorSteps = (forwardMaximum - forwardMinimum);
 
 // Calculate the nuetral range midpoint
-static const int32_t neutralMidPoint = (reverseMinimum + forwardMinimum)/2;
+static const int32_t neutralMidPoint = (reverseMinimum + forwardMinimum) / 2;
 
 // This is for showing percentages.
 static const int32_t reversePercentSteps = 100 / reverseSensorSteps;
 static const int32_t forwardPercentSteps = 100 / forwardSensorSteps;
 
+uint8_t oldSwitchVal = 1;
 // This is used to know how many samples we have in our rolling average. It's
 // main use is to deal with start-up when we don't yet have a full array.
-int32_t sampleOver       = 0;
 int32_t smoothedPotValue = 0;
 int32_t oldestPotIndex   = 0;  // This is the array index with the oldest value
 int32_t potTotal         = 0;
-int32_t potLoopCount     = 0;
 int32_t potArray[averageOver];
 
+void initializeAverages() {
+  for (int i = 0; i < averageOver; ++i) {
+    potArray[i] = neutralMidPoint;
+  }
+  potTotal = averageOver * neutralMidPoint;
+  oldestPotIndex = 0;
+}
+
 // the loop function runs over and over again forever
 void loop() {
   // Read the control switch. If this is LOW, we set the motor control voltage to 2.5v / neutral
   int32_t mainSwitchVal = digitalRead(mainSwitchPin);
 
   // Read the helm position potentiometer if the main switch is on
   int32_t sensorValue;
-  if (mainSwitchVal == 0) {
+  if (mainSwitchVal == 1) {
```

Comment:
if (mainSwitchVal == 1) is ok, 
exist a shorter version as every value greater than zero is TRUE
if (mainSwitchVal)

Reply:
Which is semantically the same but harder to understand for unexperienced programmers, which is why I put it that way.

Reply:
Thankfully for me, perl also uses the simpler if ($var) { ... } format, so I can read the shorter version. That said, future folks who might want to use this to start on their own projects might benefit from it. I assume it's no real expense, in running or compiled size?

Reply:
The created machine code should be identical.